include/roboteam_ai/control/ballHandling/BallHandlePosControl.h

```diff
@@ -50,16 +49,14 @@ class BallHandlePosControl : public NumTreePosControl {
         int ticksNotMoving = 0;
 
         pidfVals pidfGoToBall = std::make_tuple(0.0, 0.0, 0.0, 1.5);
-        PID xGoToBallPID = PID(pidfGoToBall);
-        PID yGoToBallPID = PID(pidfGoToBall);
+        PidTwoAxesController goToBallPid = PidTwoAxesController(pidfGoToBall, pidfGoToBall);
```

Comment:
@mrlukasbos The idea was to use dependency injection for the actual controller. So I have an interface (BaseController) which will be used instead of the implementation. For instance, we have here the PidTwoAxesController. Instead of it, maybe I want to use some other controller. Then I will just declare a BaseController<Vector2,Vector2>, and through some setter / in the constructor I can set which type of controller we want

Reply:
I understand. Why not set the controller you want immediately? Now you are forced to use generics making your code significantly harder.

Btw, this is very much opinion based. Its your project now and if you want it like this then don't let me stop you from doing so :-)